### PR TITLE
[release/6.0-staging] Enable loading COM component in default ALC via runtime config setting

### PR DESCRIPTION
**Pull request by elinor-fung**
Allow COM components to opt-in to being loaded in the default ALC:

- `comhost` checks for the `System.Runtime.InteropServices.COM.LoadComponentInDefaultContext` property and uses new functions on `ComActivator`, loading into the default context if the property is set to true
- Default behaviour remains loading into an isolated context
- Fall back to using `ComActivator` functions that always load into an isolated context if new functions aren't found and loading in an isolated context

Port of https://github.com/dotnet/runtime/pull/79026

## Customer impact

Forcing COM servers into isolated contexts results in many issues for any servers trying to share types. This has been a particular source of pain for customers migrating from .NET Framework and we have had a number of customers / companies indicate this has been a blocker for their migration from .NET Framework to .NET 6.

See https://github.com/dotnet/runtime/issues/66013

## Testing

Manual testing using the COMServerDemo in our samples repo. Automated tests are added for both runtime and host.

## Risk

Low. This is targeted to COM activation. There is the possibility that a user will try to use the new setting and newer comhost to run on an older runtime that does not have these changes, in which case activation of their server will fail.

### CODE REVIEW

**Comment by AaronRobinsonMSFT** [comment]
Given that 8.0 is going to be released so soon, do we see value in making this change in 6.0? I'm not as convinced of adding it here.

**Comment by elinor-fung** [comment]
I did have that thought that as well. But given the targeted nature of it, I thought it was worth at least considering in order to unblock customers that have already been trying to get from .NET Framework to .NET 6. That way, they can find any other blockers without waiting (and maybe having to deal with any 6->8 differences at the same time).

**Review by vitek-karas — APPROVED** [review]
The code change looks good.

But I do see the point in Aaron's objection. Personally I'm undecided, I see value in both ways. So I'll leave this up to triage or others if they do have strong opinions.

**Comment by carlossanlop** [comment]
@elinor-fung @vitek-karas Reminder - Tomorrow Monday 10th is Code Complete for the August Release. If you intend to get this fix included in that servicing release, please make sure to get a Tactics approval and merge the PR before 4pm, because that's when I close the branches to start merging staging into internal.

**Pull request closed without merging**
